src/lib/api/org/index.ts

```diff
@@ -275,3 +276,81 @@ function convertToSnykProject(projectData: v3ProjectData[]) : SnykProject[] {
   return projects
 }
 
+export async function listTargets(
+  requestManager: requestsManager,
+  orgId: string,
+): Promise<TargetsResponse> {
+  getApiToken();
+  getSnykHost();
+  debug(`Listing all targets for org: ${orgId}`);
+
+  if (!orgId) {
+    throw new Error(
+      `Missing required parameters. Please ensure you have set: orgId, settings.
+        \nFor more information see: https://apidocs.snyk.io/?version=2022-09-15%7Ebeta#get-/orgs/-org_id-/targets`,
+    );
+  }
+  
+  const targets = await listAllSnykTargetREST(requestManager, orgId)
+
+  return {targets};
+}
+
+export async function listAllSnykTargetREST(requestManager: requestsManager,
+  orgId: string,
+  ): Promise<SnykTargetREST[]> {
+
+    let lastPage = false;
+    let snykTargetRESTList: SnykTargetREST[] = [];
+    let pageCount = 1;
+    let nextPageLink: string | undefined = undefined;
+    while (!lastPage) {
+      try {
+        const {
+          SnykTargetREST,
+          next,
+        }: {
+          SnykTargetREST: SnykTargetREST[];
+          next?: string;
+        } = await getSnykTargetREST(requestManager, orgId, nextPageLink);
+
+        snykTargetRESTList = snykTargetRESTList.concat(SnykTargetREST);
+        next
+          ? ((lastPage = false), (nextPageLink = next))
+          : ((lastPage = true), (nextPageLink = ''));
+        pageCount++;
+      } catch (e) {
+        debug('Failed to get targets for ', orgId, e);
+        throw e;
+    }
+  }
+  return snykTargetRESTList
+}
+
+export async function getSnykTargetREST(requestManager: requestsManager,
+  orgId: string,
+  nextPageLink?: string): Promise< { SnykTargetREST: SnykTargetREST[], next?: string } > {
+
+    const url = nextPageLink ? nextPageLink : `/orgs/${orgId.trim()}/targets?version=2022-09-15~beta`
+    const res = await requestManager.request({
+    verb: 'get',
+    url: url,
+    body: undefined,
+    useRESTApi: true,
+  });
+
+  const statusCode = res.statusCode || res.status;
+  if (!statusCode || statusCode !== 200) {
+    throw new Error(
+      'Expected a 200 response, instead received: ' +
+        JSON.stringify({ data: res.data, status: statusCode }),
+    );
+  }
+    
+  const responseData = res.data
+  const SnykTargetREST = responseData.data
```

Comment:
this doesn't seem right, is it really `data.data`?  
`SnykTargetREST` should not be upper cased like this, this is for types and classes only. Please rename to `snykTarget`

Reply:
You can simplify the whole thing witj destructuring

```
const { snykTarget: data , next }  = res.data
```

Reply:
yes data.data the body we receive have data where the information are and then we need the field data of our response.